pkg/appliance/upgrade.go

```diff
@@ -0,0 +1,503 @@
+package appliance
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/appgate/sdp-api-client-go/api/v20/openapi"
+	"github.com/appgate/sdpctl/pkg/tui"
+	"github.com/appgate/sdpctl/pkg/util"
+	"github.com/cheynewallace/tabby"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-version"
+	log "github.com/sirupsen/logrus"
+)
+
+type SkipUpgrade struct {
+	Reason    error
+	Appliance openapi.Appliance
+}
+
+func (su SkipUpgrade) Error() string {
+	return fmt.Sprintf("%s: %s", su.Appliance.GetName(), su.Reason)
+}
+
+var (
+	ErrSkipReasonNotPrepared            = errors.New("appliance is not prepared for upgrade")
+	ErrSkipReasonOffline                = errors.New("appliance is offline")
+	ErrSkipReasonFiltered               = errors.New("filtered using the '--include' and/or '--exclude' flag")
+	ErrSkipReasonAlreadyPrepared        = errors.New("appliance is already prepared for upgrade with a higher or equal version")
+	ErrSkipReasonUnsupportedUpgradePath = errors.New("Upgrading from version 6.0.0 to version 6.2.x is unsupported. Version 6.0.1 or later is required.")
```

Comment:
Unrelated to PR, but caught my eyes. We should make this a parameter so the versions upgrade path is configurable in this error message.